runtime/common/src/lib.rs

```diff
@@ -738,15 +760,30 @@ macro_rules! impl_config_traits {
             type WeightInfo = weights::pallet_assets::WeightInfo<Runtime>;
         }
 
+        pub struct DustIntoTreasury;
+        type CreditOfBalances = pallet_balances::CreditOf<Runtime, ()>;
+        impl OnUnbalanced<CreditOfBalances> for DustIntoTreasury {
+            fn on_nonzero_unbalanced(mut dust: CreditOfBalances) {
+                let imbalance = NegativeImbalance::new(dust.peek());
+                Treasury::on_nonzero_unbalanced(imbalance);
+                // Ensure issuance is not reduced via OnDrop
+                core::mem::forget(dust);
```

Comment:
I think the code is correct, because the `Credit` is duplicated (because the Treasury expects a different type than what balances provides as dust). So on the one hand, `on_nonzero_unbalanced` handles a duplicate `Credit`, so whatever the treasury does not consume is reduced from total issuance when the function that handles the dust inside the treasury goes out of scope (`OnDrop`). But on the other hand, we have the original dust that was duplicated for the `on_nonzero_unbalanced` and if that goes out of scope, it reduces the whole dust from total issuance, although it potentially was already credited to the treasury through the dust duplicate via `on_nonzero_unbalanced`.

Reply:
Ah, right, didn't think about how the `new` call affects things. Cool.